tiflash: add docs (#3850)
pingcap/docs: encryption-at-rest.md

The current version of TiKV encryption has some drawbacks that we are working actively to address in the future versions.

* When a TiDB cluster is deployed, the majority of user data is stored in TiKV nodes, and that data will be encrypted when encryption is enabled. However, a small amount of user data is stored in PD nodes as metadata (for example, secondary index keys used as TiKV region boundaries). As of v4.0.0, PD doesn't support encryption at rest. It is recommended to use storage-level encryption (for example, file system encryption) to help protect sensitive data stored in PD.

> * TiFlash supports encryption at rest starting from v4.0.5. For details, refer to [Encryption at Rest for TiFlash](/encryption-at-rest.md#encryption-at-rest-for-tiflash-new-in-v405). When deploying TiKV with TiFlash before v4.0.5, data stored in TiFlash is not encrypted.

Reviewer: ```suggestion
* TiFlash supports encryption at rest since v4.0.5. For details, refer to [Encryption at Rest for TiFlash](#encryption-at-rest-for-tiflash-new-in-v405). When deploying TiKV with TiFlash earlier than v4.0.5, data stored in TiFlash is not encrypted.
```

```
./br restore full --pd <pd-address> --storage "s3://<bucket>/<prefix> --s3.region <region>"
```

## Encryption at rest for TiFlash <span class="version-mark">New in v4.0.5</span>

> TiFlash supports encryption at rest starting from v4.0.5. All files written by TiFlash (including TiFlash Proxy), including data files, schema files, temporary files, etc., are encrypted by the current data key.

Reviewer 2: The senetence "All files writtern by ..." seems duplicate with the next paragraph.

Reviewer 2: And the first two paragraph should merge to one paragraph as in its Chinese version.

Author: thx~updated. PTAL again.@lidezhu

```
./br restore full --pd <pd-address> --storage "s3://<bucket>/<prefix> --s3.region <region>"
```

## Encryption at rest for TiFlash <span class="version-mark">New in v4.0.5</span>

> TiFlash supports encryption at rest starting from v4.0.5. Data keys are generated by TiFlash. All files written by TiFlash (including TiFlash Proxy), including data files, schema files, temporary files, etc., are encrypted by the current data key. The encryption algorithms, encryption configuration (configuration items in the `tiflash-learner.toml` file) of TiFlash, and the meanings of monitoring items are consistent with those of TiKV.

Reviewer: ```suggestion
TiFlash supports encryption at rest since v4.0.5. Data keys are generated by TiFlash. All files (including data files, schema files, and temporary files) written into TiFlash (including TiFlash Proxy) are encrypted using the current data key. The encryption algorithms, the encryption configuration (in the `tiflash-learner.toml` file) supported by TiFlash, and the meanings of monitoring metrics are consistent with those of TiKV.
```

> If you deploy TiFlash with Grafana, you can look at **TiFlash-Proxy-Details** -> **Encryption**.

Reviewer: ```suggestion
If you have deployed TiFlash with Grafana, you check the **TiFlash-Proxy-Details** -> **Encryption** panel.
```

> If you have deployed TiFlash with Grafana, you check the **TiFlash-Proxy-Details** -> **Encryption** panel.

Reviewer: ```suggestion
If you have deployed TiFlash with Grafana, you can check the **TiFlash-Proxy-Details** -> **Encryption** panel.
```
